src/runtime.rs

```diff
@@ -153,7 +149,453 @@ pub struct FsSearchBody {
     pub path_group_length: u32,
 }
 
+pub struct TableGuard {
```

Comment:
It's probably worth noting that since the TableGuard's lifetime isn't tied to the lifetime of the LoadedTables it is being locked from, it can be outlived... buuuuuuuut in this case that doesn't really matter so you can probably just ignore it.